appinventor/blocklyeditor/src/blocks/helpers.js

```diff
@@ -0,0 +1,95 @@
+// -*- mode: java; c-basic-offset: 2; -*-
+// Copyright 2013-2014 MIT, All rights reserved
+// Released under the Apache License, Version 2.0
+// http://www.apache.org/licenses/LICENSE-2.0
+
+
+/**
+ * @license
+ * @fileoverview Helper block definitions. These are created based on the
+ *     component database, and are meant to be connected to component
+ *     method/setter inputs to make coding faster for users.
+ */
+
+'use strict';
+
+goog.provide('Blockly.Blocks.helpers');
+
+goog.require('Blockly.Blocks.Utilities');
+
+Blockly.COLOUR_HELPERS = "#BF4343";
+
+Blockly.Blocks['helpers_dropdown'] = {
+  init: function() {
+    /**
+     * The key to the OptionList associated with this block.
+     * @private
+     */
+    this.key_ = "";
+
+    // TODO: Do we want to do anything for tooltips on this guy?
+    this.setColour(Blockly.COLOUR_HELPERS);
+    // Everything else gets hander by domToMutaiton.
+  },
+ 
+  mutationToDom: function() {
+    var mutation = document.createElement('mutation');
+    mutation.setAttribute('key', this.key_);
+    return mutation;
+  },
+
+  domToMutation: function(xml) {
+    this.key_ = xml.getAttribute('key');
+    var type = Blockly.Blocks.Utilities.helperKeyToBlocklyType(
+      { type: 'OPTION_LIST', key: this.key_ }, this);
+
+    var optionList = this.getTopWorkspace().getComponentDatabase()
+        .getOptionList(this.key_);
+
+    var dropdown = new Blockly.FieldInvalidDropdown(
+        this.getValidOptions(), this.getInvalidOptions());
+
+    this.setOutput(true, type);
```

Comment:
Output types are currently only used for blockly-side connection checking. The only types that get encoded into yail are input types, so that arguements can be coerced before being passed to Java / Yail primitives.

Added [here](https://github.com/BeksOmega/appinventor-sources/pull/5/files#diff-7aa12b3497a5f408877d090b517399d8R50).